nbgrader/tests/apps/test_nbgrader_db.py

```diff
@@ -176,6 +176,61 @@ def test_student_import(self, db, temp_cwd):
             assert student.first_name is None
             assert student.email is None
 
+
+    def test_student_import_csv_spaces(self, db, temp_cwd):
+        with open("students.csv", "w") as fh:
+            fh.write(dedent(
+                """
+                id,first_name,last_name, email
+                foo,abc,xyz,foo@bar.com
+                bar,,,
+                """
+            ).strip())
+
+        run_nbgrader(["db", "student", "import", "students.csv", "--db", db])
+        with Gradebook(db) as gb:
+            student = gb.find_student("foo")
+            assert student.last_name == "xyz"
+            assert student.first_name == "abc"
+            assert student.email == "foo@bar.com"
+            student = gb.find_student("bar")
+            assert student.last_name is None
+            assert student.first_name is None
+            assert student.email is None
+
+        # check that it fails when no id column is given
```

Comment:
Is anything after this line different from what's in `test_student_import`? If not I think you can delete it.